bool front
bool tail
---
string result